haas/ext/auth/database.py

```diff
@@ -0,0 +1,147 @@
+"""Auth plugin using usernames & passwords in the DB, with HTTP basic auth.
+
+Includes API calls for managing users.
+"""
+from haas import api, model, auth
+from haas.auth import get_auth_backend
+from haas.rest import rest_call, local
+from haas.errors import *
+from sqlalchemy import Column, ForeignKey, String, Boolean, Table
+from sqlalchemy.orm import relationship
+from passlib.hash import sha512_crypt
+from schema import Schema, Optional
+import flask
+
+
+class User(model.Model):
+    """A user of the HaaS.
+
+    A user can be a member of any number of projects, which grants them access
+    to that process's resources. A user may also be flagged as an administrator.
+    """
+
+    is_admin = Column(Boolean, nullable=False, default=False)
+    # The user's salted & hashed password. We currently use sha512 as the
+    # hashing algorithm:
+    hashed_password = Column(String)
+
+    # The projects of which the user is a member.
+    projects = relationship('Project', secondary='user_projects', backref='users')
+
+    def __init__(self, label, password, is_admin=False):
+        """Create a user `label` with the specified (plaintext) password."""
+        self.label = label
+        self.is_admin = is_admin
```

Comment:
Also, I will point out that this isn't really part of this pr; there's a lot of stuff here that's really part of #491, but since that hasn't actually been merged, it's showing up here as well.